### ISSUE TITLE
Exception: Same key has already been added

### ISSUE THREAD

**Issue by aongeek** [OfficeDev/Office-Inspectors-for-Fiddler#19]
Hi,
I am seeing this error which is rather less frequent.

System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at MAPIInspector.Parsers.ROPInputBuffer.Parse(Stream s)
   at MAPIInspector.Parsers.rgbInputBuffer.Parse(Stream s)
   at MAPIInspector.Parsers.ExecuteRequestBody.Parse(Stream s)
   at MapiInspector.MAPIInspector.ParseHTTPPayload(HTTPHeaders headers, Int32 currentSessionID, Byte[] bytesFromHTTP, TrafficDirection direction, Byte[]& bytes)

Here is fiddler dump:
https://www.dropbox.com/s/fcgye3q8kphiyw8/duplicate_item.saz?dl=0

Thanks,
Aon

**Comment by tarunchopra** [comment]
Hello Aongeek

Thanks for reporting the issue. We are looking into it.

^TC

**Comment by tarunchopra** [comment]
Hello Aongeek

We have checked in the fix based on code review since we are unable to reproduce the issue. It will be helpful if you can verify the fix or share the repro.

^TC.

**Comment by aongeek** [comment]
Hi Tarun, 

I still see errors. I noticed that it mostly happens when you launch outlook and it try to sync your messages. 

System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at MAPIInspector.Parsers.ROPInputBuffer.Parse(Stream s)
   at MAPIInspector.Parsers.rgbInputBuffer.Parse(Stream s)
   at MAPIInspector.Parsers.ExecuteRequestBody.Parse(Stream s)
   at MapiInspector.MAPIInspector.ParseHTTPPayload(HTTPHeaders headers, Int32 currentSessionID, Byte[] bytesFromHTTP, TrafficDirection direction, Byte[]& bytes)

Thanks.
